src/core/gui/Builder.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Xournal++
+ *
+ * gtk_builder wrapper to parse ui files
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ */
+
+#pragma once
+
+#include <cassert>
+#include <iostream>
+#include <string>
+
+#include <gtk/gtk.h>
+
+#include "util/raii/GObjectSPtr.h"
+
+class GladeSearchpath;
+
+class Builder final {
+public:
+    Builder(GladeSearchpath* gladeSearchPath, const std::string& uiFile);
+    ~Builder();
+
+    template <class GObjectType>
+    GObjectType* get(const std::string& name) {
```

Comment:
May it be a good Idea to also return a guarded pointer like `GObjectSPtr<GObjectType>`?